src/core/net/dnssd_server.hpp

```diff
@@ -570,6 +572,8 @@ class Server : public InstanceLocator, private NonCopyable
 
         static bool IsActionForAdditionalSection(ProxyAction aAction, const Questions &aQuestions);
 
+        static constexpr uint32_t kMaxTtlInSeconds = 10; // RFC 8766 Section 5.5.1 and Section 5.6
```

Comment:
Since TTL is always in seconds, suggest maybe renaming:
```suggestion
        static constexpr uint32_t kMaxTtl = 10; // RFC 8766 Section 5.5.1 and Section 5.6
```